src/pinnable_mapped_file.cpp

```diff
@@ -156,8 +175,16 @@ pinnable_mapped_file::pinnable_mapped_file(const bfs::path& dir, bool writable,
       });
 
       try {
-         setup_non_file_mapping();
-         load_database_file(sig_ios);
+         auto existing_file_size = bfs::file_size(_data_file_path);
+         if (existing_file_size) {
+            setup_non_file_mapping(existing_file_size);
+            load_database_file(sig_ios);
+         }
+         else {
+            setup_non_file_mapping(shared_file_size);
+            file_mapped_segment_manager = new ((char*)_non_file_mapped_mapping+header_size) segment_manager(shared_file_size-header_size);
+            new (_non_file_mapped_mapping) db_header;
```

Comment:
Are these just leaked on exit?

Reply:
No, this is a placement new, which construct the object in a known memory address (in this case, its the mmap() returned address), it cannot be deleted simply because it doesn't allocate memory by itself, the memory is actually deallocated by munmap().